tools/snp.sh

```diff
@@ -1081,42 +1062,134 @@ attest_guest() {
   ssh_guest_command "sudo insmod /lib/modules/*/kernel/drivers/virt/coco/sev-guest/sev-guest.ko >/dev/null 2>&1 || true"
 
   # Request and display the snp attestation report with random data
-  ssh_guest_command "sudo ./snpguest report attestation-report.bin request-data.txt --random"
-  ssh_guest_command "./snpguest display report attestation-report.bin"
+  if [ "$SVSM" = true ]; then
+	ssh_guest_command "sudo ./snpguest report attestation-report.bin request-data.txt --random"
+  else
+	ssh_guest_command "sudo ./$Home/snpguest/target/release/snpguest report attestation-report.bin request-data.txt --random --vmpl 3"
+  fi
+  ssh_guest_command "./$Home/snpguest/target/release/snpguest display report attestation-report.bin"
 
   # Retrieve ark, ask, vcek (saved in ./certs)
-  ssh_guest_command "./snpguest fetch ca pem ${cpu_code_name} . --endorser vcek"
-  ssh_guest_command "./snpguest fetch vcek pem ${cpu_code_name} . attestation-report.bin"
+  ssh_guest_command "./$Home/snpguest/target/release/snpguest fetch ca pem ${cpu_code_name} ."
+  ssh_guest_command "./$Home/snpguest/target/release/snpguest fetch vcek pem ${cpu_code_name} . attestation-report.bin"
```

Comment:
This seems like some very different behavior from what we were doing previously. Is there a reason we are building the tool within the guest instead of on the host machine and copying it over? @ryansavino  thoughts?

Reply:
Got it. So, installing and running snpguest directly in the guest VM ensures compatibility with the glibc and tools available on that VM, avoiding issues caused by differences between the host(ubuntu 24.04) and guest (ubuntu 22.02) environments. If you need any further adjustments or have more questions, just let me know.